src/lossy_compression/value_representation/serf_qt.zig

```diff
@@ -0,0 +1,325 @@
+/// `decompressed_values`. The `allocator` is used to manage the memory of intermediate results.
+/// If an error occurs it is returned.
+pub fn decompress(
+    allocator: Allocator,
+    compressed_values: []const u8,
+    decompressed_values: *ArrayList(f64),
+) Error!void {
+    // Ensure the compressed values are not empty, i.e., at least the header is present.
+    if (compressed_values.len < 4) return Error.UnsupportedInput;
+
+    // Read bucket_size from the header.
+    const bucket_size: f32 = @bitCast(compressed_values[0..4].*);
+
+    var decoded_quantized_values = ArrayList(u64).init(allocator);
+    defer decoded_quantized_values.deinit();
+
+    // Decode the Elias Gamma encoded quantized values.
+    try shared_functions.decodeEliasGamma(
+        compressed_values[4..compressed_values.len],
+        &decoded_quantized_values,
+    );
+
+    // Reconstruct each value from its quantized representation.
+    var previous_val: f64 = 0.0;
+    for (decoded_quantized_values.items) |quantized_value| {
+        var decompressed_value: f64 = 0.0;
+        if (bucket_size == 0.0) {
+            // If bucket size is zero, we assume the values were not quantized and are stored as usize directy.
+            decompressed_value = shared_functions.orderedBitsToFloat(quantized_value);
+        } else {
+            // Reconstruct value from quantized_value and append to decompressed_value.
+            const un_zigzag_value: f64 = @floatFromInt(shared_functions.decodeZigZag(quantized_value - 1));
+            decompressed_value = previous_val + un_zigzag_value * bucket_size;
+            previous_val = decompressed_value;
+        }
+        try decompressed_values.append(decompressed_value);
+    }
+}
+
+test "serfqt can compress and decompress bounded values" {
+    const allocator = testing.allocator;
+    const data_distributions = &[_]tester.DataDistribution{
+        .LinearFunctions,
+        .BoundedRandomValues,
+        .SinusoidalFunction,
+        .MixedBoundedValuesFunctions,
+    };
+
+    try tester.testErrorBoundedCompressionMethod(
+        allocator,
+        Method.SerfQT,
+        data_distributions,
+    );
+}
+
+test "serfqt cannot compress and decompress nan values" {
+    const allocator = testing.allocator;
+    const uncompressed_values = [3]f64{ 343.0, math.nan(f64), 520.0 };
+    var compressed_values = ArrayList(u8).init(allocator);
+    defer compressed_values.deinit();
+
+    const method_configuration =
+        \\ {"abs_error_bound": 0.1}
+    ;
+
+    compress(
+        allocator,
+        uncompressed_values[0..],
+        &compressed_values,
+        method_configuration,
+    ) catch |err| {
+        try testing.expectEqual(Error.UnsupportedInput, err);
+        return;
+    };
+
+    try testing.expectFmt(
+        "",
+        "The SerfQT method cannot compress nan values",
+        .{},
+    );
+}
+
+test "serfqt cannot compress and decompress unbounded values" {
```

Comment:
Unclear what bounded and unbounded values means in the tests, maybe add comments to document it.